core/write.go

```diff
@@ -202,7 +212,34 @@ func copy(t reflect.Value, v reflect.Value) (err error) {
 		return
 	}
 
-	t.Set(v)
+	// if we are copying from one slice or array to another
+	if isList(v) && isList(t) {
+		// loop over every item in the desired value
+		for i := 0; i < v.Len(); i++ {
+			// write to the target given its kind
+			switch t.Kind() {
+			// if its a slice
+			case reflect.Slice:
+				// an object of the correct type
+				obj := reflect.Indirect(reflect.New(t.Type().Elem()))
+
+				// write the appropriate value to the obj and catch any errors
+				if err := copy(obj, v.Index(i)); err != nil {
+					return err
+				}
+
+				// just append the value to the end
+				t.Set(reflect.Append(t, obj))
```

Comment:
Not sure, I wonder if the entire `reflect.Slice` section can be simplified to basically match you do in the `reflect.Array` case.  Perhaps something like:
```
case reflect.Slice:
    if t.Cap() < v.Len() {
        t.SetCap(v.Len())
    }
    fallthrough
case reflect.Array:
    copy(t.Slice(i, i+1).Index(0), v.Index(i))
```